Add Feature/audit and fix postgresql reserved keyword error

LazaroKy opened the pull request:
### Add AuditModel and apply auditing to entities

```java
AuditModel {  
    Long id;  
    Instant createdAt;  
    Instant lastModifiedAt;  
    Long createdBy;  
    Long lastUpdatedBy;  
   }
```

### Implemented AuditorAware to retrieve the current user (by email) for auditing.

### Fixed PostgreSQL error caused by reserved keyword user by renaming the table to users.

- closes (#58)

LazaroKy committed:
feat: Add Audit Model

LazaroKy committed:
feat: implement audit for entities

LazaroKy committed:
feat: add AuditorAware implementation to retrieve auditor

LazaroKy committed:
fix: rename User table to 'users' to fix PostgreSQL reserved keyword error

Julio573 reviewed (approved):
Gostei muito da maneira que você isolou a responsabilidade de auditoria, deixando o código mais desacoplado. Não acho que precise de melhorias, foi uma ótima adição!

KayoPereira reviewed (changes requested):
Na entidade created_by, já estamos tratando a possibilidade de contem um id e um texto? Se a coluna é referência (FK) não pode conter texto que não seja uma key. logo podemos pensar em até usar a abordagem de se contem user_id no created_by o user estava logado e se não tiver nada na coluna, a auditoria foi gerada sem o usuário está logado e o campo é NULL TRUE.

Comentem sobre isso, quero ver a opinião de todos. @LazaroKy @vinnyhbr @Julio573 @petrosbarreto

LazaroKy committed:
refactor: return 'guestUser' for unauthenticated users in auditor retrieval

Need to define user authentication method - JWT, UserDetails, OAuth,.. - as it will affect the return of the Principal
